src/main/java/seedu/contax/logic/commands/RangeCommand.java

```diff
@@ -0,0 +1,114 @@
+
+/**
+ * Range edit or delete a person identified using it's displayed from index and to index from the address book.
+ */
+public class RangeCommand extends Command {
+    public static final String COMMAND_WORD = "range";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": command in range"
+            + "by the index number used in the displayed person list. "
+            + "Parameters: from/FROM_INDEX to/TO_INDEX (must be a positive integer) "
+            + "Example command range edit and range delete ";
+
+    public static final String MESSAGE_EDIT_USAGE = COMMAND_WORD + ": Edits the details of the person in range"
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "[" + PREFIX_PHONE + "PHONE] "
+            + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Parameters: from/FROM_INDEX to/TO_INDEX (must be a positive integer) "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_PHONE + "91234567 "
+            + PREFIX_EMAIL + "johndoe@example.com"
+            + PREFIX_RANGE_FROM + "1"
+            + PREFIX_RANGE_TO + "3";
+
+    private final Index fromIndex;
+    private final Index toIndex;
+    private final ArgumentMultimap argMultimap;
+    private final String commandInput;
+    /**
+     * @param fromIndex              of the person in the filtered person list to edit
+     * @param toIndex                of the person in the filtered person list to edit
+     * @param argMultimap            details to argMultimap
+     * @param commandInput            details to word of command
+     */
+    public RangeCommand(Index fromIndex, Index toIndex, ArgumentMultimap argMultimap, String commandInput) {
+        requireNonNull(fromIndex);
+        requireNonNull(toIndex);
+        requireNonNull(argMultimap);
+        this.fromIndex = fromIndex;
+        this.toIndex = toIndex;
+        this.argMultimap = argMultimap;
+        this.commandInput = commandInput.trim();
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        if (fromIndex.getZeroBased() > toIndex.getZeroBased()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+        List<CommandResult> commandResultList = new ArrayList<>();
+        for (int i = fromIndex.getOneBased(); i <= toIndex.getOneBased(); i++) {
+            AddressBookParser addressBookParser = new AddressBookParser();
+            try {
+                String commandText = createNewCommand(commandInput, Integer.toString(i));
+                if (argMultimap.getPreamble().equals("delete")) {
+                    commandText = createNewCommand(commandInput, Integer.toString(fromIndex.getOneBased()));
+                }
```

Comment:
This could have been a if else, but more importantly, there is no need to separate out the delete case.

Reply:
thx for the suggestion, but unlike edit, delete will modify index, as execute command in sequence, the index is different from edit i.e. always be the from index delete until to

Reply:
Then how about performing the operation in reverse order? Then, your indexes will not change

Reply:
yah good point I will try